### ISSUE TITLE
Use Babel for main process

### ISSUE THREAD

**Issue by davej** [chentsulin/electron-react-boilerplate#139]
What's the consensus about letting the main process run through babel too? The issue I'm having is that I want to use `redux-electron-store` but I can't require any of the reducers because they use the ES6 module syntax which currently isn't supported by the latest versions of Electron/Chromium.

**Comment by jhen0409** [comment]
Yes,  if we don‘t use Babel, this is no way to share reducers between Main and Renderer.

@chentsulin what do you think?

**Comment by chentsulin** [comment]
I don't mind using babel also in main process, but which one approach is our best choice? `babel-register`?

**Comment by davej** [comment]
The problem with using babel-register is that babel then needs to be bundled with the app (rather than simply as a dev-dependency).

**Comment by davej** [comment]
Actually, I guess we could use `babel-register` in development and then compile it (using webpack?) for the production version.

**Comment by wasd171** [comment]
Code sharing is quite essential. Unfortunately, simple adding of `require('babel-register');` in `main.js` results in 
```
App threw an error when running TypeError: Path must be a string. Received undefined
```
requiring the following `dummy.js`:
```
module.exports = 'bar';
```
I've tried webpack'ing `main.js`, but no luck :(

**Comment by naderhen** [comment]
Is there a known workaround for this? @davej your solution seems reasonable, but I'm curious how you'd go about it within this boilerplate.

**Comment by davej** [comment]
@naderhen: it just worked for me.

My entry point file looks like this:

```js
/* eslint strict: 0 */
'use strict';

require('babel-register');
require('./main/init.js');
```

`./main/init.js` is where my `main` process logic is.

**Comment by naderhen** [comment]
I have something very similar which does indeed work in terms of running the electron application with babel included. However, when I attempt to package the app via `npm run package` (on OSX) and run the resulting *.app I get a host of babel related import errors:

![image](https://cloud.githubusercontent.com/assets/99392/14397101/80c55170-fda9-11e5-807c-0f27c36c539d.png)

Any thoughts? Do you run into the same issue when packaging?

**Comment by davej** [comment]
@naderhen: I haven't tried packaging the app yet. My plan was to use webpack and create an entry for the main js file. Have you tried using webpack to compile the main process files instead of including babel in your production build?

**Comment by naderhen** [comment]
hmm, yea @davej I've given the separate webpack entry route a shot but ended up with a slew of other cryptic errors either during the build or packaging process. If you come up with a working solution on your end I'd love to be pointed in the right direction. Willing to test out any other options you think would work too.

**Comment by chentsulin** [comment]
Closed via #197.
